components/kyma-environment-broker/internal/avs/delegator.go

```diff
@@ -0,0 +1,117 @@
+package avs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-incubator/compass/components/provisioner/pkg/gqlschema"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	overrideKey     = "avs_bridge.config.evaluations.cluster.id"
+	avsBridgeAPIKey = "avs_bridge.config.availabilityService.apiKey"
+)
+
+type delegator struct {
+	operationManager *process.OperationManager
+	avsConfig        Config
+	clientHolder     *clientHolder
+}
+
+func newDelegator(avsConfig Config, operationsStorage storage.Operations) *delegator {
+	return &delegator{
+		operationManager: process.NewOperationManager(operationsStorage),
+		avsConfig:        avsConfig,
+		clientHolder:     newClientHolder(avsConfig),
+	}
+}
+
+func (del *delegator) doRun(logger logrus.FieldLogger, stepName string, operation internal.ProvisioningOperation,
+	modelSupplier func(provisioningOperation internal.ProvisioningOperation) (*basicEvaluationCreateRequest, error)) (internal.ProvisioningOperation, time.Duration, error) {
+	logger.Infof("starting the step %s", stepName)
+
+	if operation.AvsEvaluationInternalId != 0 {
+		msg := fmt.Sprintf("%s has already been finished previously", stepName)
+		return del.operationManager.OperationSucceeded(operation, msg)
+	}
+
+	evaluationObject, err := modelSupplier(operation)
+	if err != nil {
+		logger.Errorf("Step %s failed with error %v", stepName, err)
+		return del.operationManager.OperationFailedWithError(operation, err)
+	}
+
+	evalResp, err := del.postRequest(evaluationObject, logger)
+	if err != nil {
+		return del.operationManager.OperationFailedWithError(operation, err)
```

Comment:
are you sure you want to break provisioning kyma in case of any error? for example temporary network problem